### PR TITLE
Fix BPF policy map specification inconsistency between BPF programs

### PR DESCRIPTION
**Pull request by joestringer**
Three paths were defining and using a POLICY_MAP with the same helper
functions, consolidate the declaration in one place. This fixes the
following issue reported by André when switching between `--lb` and
`--device` mode.

    + tc filter add dev cilium_vxlan ingress prio 1 handle 1 bpf da obj bpf_overlay.o sec from-overlay
    Map specification differs from pinned file!
     - Size key:     4 (obj) != 8 (pin)
    Map 'cilium_policy_reserved_2' self-check failed!
    Error loading maps into kernel!
    Error fetching ELF ancillary data!
    Note: 8 bytes struct bpf_elf_map fixup performed due to size mismatch!

Signed-off-by: Joe Stringer <joe@covalent.io>

### CODE REVIEW

**Comment by joestringer** [comment]
test-me-please

File: bpf/netdev_config.h [hunk 1]
```diff
@@ -25,3 +25,4 @@
 #define HANDLE_NS
 #define FROM_HOST
 #define ENCAP_IFINDEX 1
+#define POLICY_MAP cilium_policy_foo
```
Comment (ianvernon) [inline bpf/netdev_config.h:28]: Why "foo"?

Comment (joestringer) [reply bpf/netdev_config.h:28]: This config file is only used during local builds, and this ensures that the POLICY_MAP name is defined to be "something". In actual deployments, `cilium-agent` will define this to be something sane when generating the BPF header files.

**Review by borkmann — APPROVED** [review]
lgtm

**Pull request merged**